src/firmware/app/control/trajectory_planner.h

```diff
@@ -11,293 +13,434 @@ typedef struct FirmwareRobotPathParameters
- * trajectory
- * @pre traj_elements & velocity_profile are pre-allocated up to at least
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ * This function is intended to take a variable time trajectory (which is easier to
+ * calculate) then interpolate a constant period trajectory. This is valuable for discrete
+ * time motion controllers that operate only at constant delta-time intervals
  *
- * @param traj_elements [in/out] The trajectory element array that will be modified to
- * contain the time profile
- * @param num_segments [in] The number of segments(elements) in traj_elements and
- * velocity_profile
- * @param arc_segment_length  [in] The arc length of each path segment
- * @param velocity_profile [in] The forwards and backwards continuous velocity profile of
- * the trajectory
+ * NOTE: This function does not modify the real trajectory. It will output a physically
+ * invalid trajectory if the input trajectory is invalid
+ *
+ * @param constant_period_trajectory [out] The generated constant-period trajectory that
+ * will be linearly interpolated from a variable-period trajectory.
+ *
+ * @param variable_period_trajectory [in] The variable-period trajectory that will be used
+ * to generate the constant-period trajectory.
+ *
+ * @param num_elements [in/out] The number of elements in the trajectory. The num_elements
+ * parameter will be modified to contain the num_elements of the new constant-period
+ * trajectory generated.
+ *
+ * @param interpolation_period [in] The interpolation period of the constant-period
+ * trajectory. This parameter defines the length of time between successive elements in
+ * the trajectory. In seconds.
+ *
+ * @return A status indicating whether or not generation was successful
  */
-void static app_trajectory_planner_generatePositionTrajectoryTimeProfile(
-    PositionTrajectoryElement_t* traj_elements, const float num_segments,
-    const float arc_segment_length, float* velocity_profile);
+TrajectoryPlannerGenerationStatus_t
+app_trajectory_planner_interpolateConstantPeriodPositionTrajectory(
+    PositionTrajectory_t* constant_period_trajectory,
+    PositionTrajectory_t* variable_period_trajectory, unsigned int* num_elements,
+    float interpolation_period);
 
-/***
- *  This function takes in a forwards trajectory annd modifies it in place to become a
- * backwards trajectory NOTE: This function exists to avoid issues outlined in #1322
+/**
+ * Function generates a constant interpolation period trajectory from the path parameters
+ * specified in the constant parameterization trajectory. The function creates a constant
+ * parameterization trajectory then uses linear interpolation to calculate the equivalent
+ * constant interpolation period trajectory
+ *
+ * NOTE: Constant interpolation period means that the duration of time between each point
+ * on the trajectory is CONSTANT. This is different from a constant parameterization
```

Comment:
we use `constantPeriod` above and `constantInterpolationPeriod` here. Are they the same? If so, we should choose one to stick with, if not, we should make sure we're documenting the difference clearly (I think I get the difference, but we really want to avoid any ambiguity here).

Reply:
They are the same - I've updated everything to use 'constant period'